pkg/block/fetcher.go

```diff
@@ -301,33 +412,42 @@ func (f *BaseFetcher) fetchMetadata(ctx context.Context) (interface{}, error) {
 	for i := 0; i < f.concurrency; i++ {
 		eg.Go(func() error {
 			for id := range ch {
-				meta, err := f.loadMeta(ctx, id)
-				if err == nil {
-					mtx.Lock()
-					resp.metas[id] = meta
-					mtx.Unlock()
-					continue
+				meta, metaErr := f.loadMeta(ctx, id)
+
+				var (
+					mark    *cachedDeletionMark
+					markErr error
+				)
+				if metaErr == nil {
+					mark, markErr = f.loadDeletionMark(ctx, id, now)
 				}
```

Comment:
Thinking loudly: the contract of this `fetchMetadata()` function is that it returns found metas also in case of any error fetching any meta (so it's OK returning a partial view). I'm wondering if it's correct to never fetch the deletion mark in case of any error: ie. should we fetch it anyway in case of `ErrorSyncMetaCorrupted`? I don't have an answer honestly, but I would like to have a discussion on it.